packages/repository/src/repositories/lyric/repository.ts

```diff
@@ -0,0 +1,135 @@
+import * as path from 'path';
+
+import {
+	type DataRecordValue,
+	type IRepository,
+	logger,
+	type LyricRepositoryConfig,
+} from '@overture-stack/maestro-common';
+
+import httpClient from '../../network/httpClient';
+import { isArrayOfObjects } from '../../utils/utils';
+
+/**
+ * Implementation to use Lyric as a data repository
+ * @param config
+ * @returns
+ */
+export const lyricRepository = (config: LyricRepositoryConfig): IRepository => {
+	const { baseUrl, categoryId, paginationSize } = config;
+	const getRepositoryRecords = async function* (): AsyncGenerator<Record<string, DataRecordValue>[], void, unknown> {
+		let page = 1;
+		let hasMoreData = true;
+
+		// Get all records paginated by category
+		// http://lyric/data/category/{categoryId}?view=compound&page={pageNumber}&pageSize={pageSize}
+		const fullUrl = new URL(path.join('data', 'category', categoryId.toString()), baseUrl);
+		fullUrl.searchParams.append('view', 'compound');
```

Comment:
Might want to make these string literals into constant variables that we reference. This search param line is reused further down this file.